pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -257,3 +265,53 @@ func (c *cephStatusChecker) conditionMessageReason(condition cephv1.ConditionTyp
 
 	return condition, reason, message
 }
+
+// forceDeleteStuckPodsOnNotReadyNodes lists all the nodes that are in NotReady state and
+// gets all the pods on the failed node and force delete the pods stuck in terminating state.
+func (c *cephStatusChecker) forceDeleteStuckRookPodsOnNotReadyNodes() error {
+	nodes, err := k8sutil.GetNotReadyKubernetesNodes(c.context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to get NotReady nodes")
+	}
+	for _, node := range nodes {
+		pods, err := c.getRookPodsOnNode(node.Name)
+		if err != nil {
+			logger.Errorf("failed to get pods on NotReady node %q. %v", node.Name, err)
+		}
+		for _, pod := range pods {
+			if err := k8sutil.ForceDeletePodIfStuck(c.context, pod); err != nil {
+				logger.Warningf("skipping forced delete of stuck pod %q. %v", pod.Name, err)
+			}
+		}
+	}
+	return nil
+}
+
+func (c *cephStatusChecker) getRookPodsOnNode(node string) ([]v1.Pod, error) {
+	clusterName := c.clusterInfo.NamespacedName()
+	appLabels := []string{
+		"csi-rbdplugin-provisioner",
+		"csi-rbdplugin",
+		"csi-cephfsplugin-provisioner",
+		"csi-cephfsplugin",
+		"rook-ceph-operator",
+	}
+	podsOnNode := []v1.Pod{}
+	pods, err := c.context.Clientset.CoreV1().Pods(clusterName.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return podsOnNode, errors.Wrapf(err, "failed to get pods on node %q", node)
+	}
+	for _, pod := range pods.Items {
+		if pod.Labels["rook_cluster"] == clusterName.Name {
+			podsOnNode = append(podsOnNode, pod)
+			continue
+		}
+		for _, label := range appLabels {
+			if pod.Labels["app"] == label {
+				podsOnNode = append(podsOnNode, pod)
```

Comment:
I think there may need to be a `break` statement here to break the `for _, label := range appLabels` to prevent a pod from being added multiple times. Once we determine a pod needs to go in the list, we don't need to do further processing on other labels.

Reply:
added